sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.controller.service.ts

```diff
@@ -57,6 +62,39 @@ export class ScholasticStandingControllerService {
     };
   }
 
+  /**
+   * Get Scholastic Standing summary details.
+   * @param studentId student id to retrieve the scholastic standing summary.
+   * @returns Scholastic Standing summary details.
+   */
+  async getScholasticStandingSummary(
+    studentId: number,
+  ): Promise<ScholasticStandingSummaryDetailsAPIOutDTO> {
+    const studentExists = await this.studentService.studentExists(studentId);
+    if (!studentExists) {
+      throw new NotFoundException("Student does not exists.");
+    }
+    const scholasticStandingSummaryPromise =
+      this.studentScholasticStandingsService.getScholasticStandingSummary(
+        studentId,
+      );
+    const sfasUnsuccessfulCompletionWeeksPromise =
+      this.sfasIndividualService.getSFASTotalUnsuccessfulCompletionWeeks(
+        studentId,
+      );
+    const [scholasticStandingSummary, sfasUnsuccessfulCompletionWeeks] =
+      await Promise.all([
+        scholasticStandingSummaryPromise,
+        sfasUnsuccessfulCompletionWeeksPromise,
+      ]);
+    const totalUnsuccessfulCompletionWeeks =
+      +(scholasticStandingSummary?.totalUnsuccessfulWeeks ?? 0) +
```

Comment:
why is + added for typecasting here?

Reply:
There is one more related comment about it.